libs/cache/file_cache.go

```diff
@@ -0,0 +1,286 @@
+package cache
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sync"
+	"time"
+
+	"github.com/databricks/cli/internal/build"
+	"github.com/databricks/cli/libs/env"
+	"github.com/databricks/cli/libs/log"
+)
+
+// Metrics is a local interface for tracking cache telemetry.
+type Metrics interface {
+	SetBoolValue(key string, value bool)
+	SetDurationValue(key string, value time.Duration)
+}
+
+// FileCache implements the Cache interface using local disk storage.
+type FileCache[T any] struct {
+	baseDir       string
+	expiryMinutes int
+	mu            sync.Mutex
+	metrics       Metrics
+	cacheEnabled  bool // If true, cached values are returned; if false, cache is only used for measurement
+}
+
+// newFileCacheWithBaseDir creates a new file-based cache that stores data in the specified directory.
+func newFileCacheWithBaseDir[T any](ctx context.Context, baseDir string, expiryMinutes int) (*FileCache[T], error) {
+	if err := os.MkdirAll(baseDir, 0o700); err != nil {
+		return nil, fmt.Errorf("failed to create cache directory: %w", err)
+	}
+
+	fc := &FileCache[T]{
+		baseDir:       baseDir,
+		expiryMinutes: expiryMinutes,
+	}
+
+	// Clean up expired files synchronously
```

Comment:
Not sure if that's the best - we use cache to quickly look up things but we also run potentially expensive clean up synchronously before we can use it.